spec/serializers/provider_serializer_spec.rb

```diff
@@ -35,4 +35,33 @@
   it { should include(postcode: provider.enrichments.last.postcode) }
   it { should include(institution_type: provider.provider_type) }
   it { should include(accrediting_provider: nil) }
+
+  describe 'ProviderSerializer#region_code' do
+    subject do
+      serialize(provider)["region_code"]
+    end
+
+    describe "provider region code 'London' can be overriden by enrichment region code 'Scotland'" do
+      let(:enrichment) do
+        build(:provider_enrichment, region_code: "Scotland")
+      end
+
+      let(:provider) { create :provider, region_code: "London", enrichments: [enrichment] }
+      it { is_expected.not_to eql(format("%02d", 1)) }
+      it { is_expected.to eql(format("%02d", 11)) }
+      it { expect(subject.length).to eql(2) }
+    end
+
+    region_code = 1
+    describe "provider region code 00 is overriden with #{region_code} " do
+      let(:enrichment) do
+        build(:provider_enrichment, region_code: region_code)
+      end
```

Comment:
Same as above here, but also I would not use `region_code` here as a variable, the way that's usually done here is with a let:

```
describe 'overriding region_code' do
  let(:enrichment) do
    build(:provider_enrichment, region_code: region_code)
  end
  let(:region_code) { 1 }

  ...
end
```